### PR TITLE
cmp not treating 0 as integer only as string

### PR DESCRIPTION
**Pull request by jeremymv2**
This change allows both tests to pass, which I believe is the intent of the `cmp` matcher.

```
describe 0 do
  it { should cmp 0 }
  it { should cmp '0' }
end
```

Without the change from this PR, it fails one of the tests:

```
~/Devel/ChefProject/inspec (cmp_treat_0_as_integer $%=)$ inspec exec ../tmp/cmp_test.rb --format=documentation

0
  should cmp 0
  should cmp "0" (FAILED - 1)

Failures:

  1) 0 should cmp "0"
     Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }

       expected: "0"
            got: 0

       (compared using `cmp` matcher)
     # ../tmp/sshd_test.rb:7:in `block (2 levels) in load'

Finished in 0.01047 seconds (files took 0.79418 seconds to load)
2 examples, 1 failure

Failed examples:

rspec  # 0 should cmp "0"

~/Devel/ChefProject/inspec (cmp_treat_0_as_integer $%=)$
```

### CODE REVIEW

File: lib/matchers/matchers.rb [hunk 1]
```diff
@@ -232,7 +232,7 @@
 RSpec::Matchers.define :cmp do |first_expected|
 
   def integer?(value)
-    !(value =~ /\A[1-9]\d*\Z/).nil?
+    !(value =~ /\A[0-9]\d*\Z/).nil?
```
Comment (chris-rock) [inline lib/matchers/matchers.rb]: Great catch and really good test-case. We left 0 out, to cover the detection for octal numbers. Therefore 0 is only valid if the length of the string is 1

Comment (chris-rock) [inline lib/matchers/matchers.rb]: Great catch and really good test-case. We left 0 out, to cover the detection for octal numbers. Therefore 0 is only valid if the length of the string is 1

**Comment by jeremymv2** [comment]
@chris-rock assuming that last commit passes tests, could you lend 👀 on this please to ensure it's sane for what how you expect 'cmp' to work?

**Comment by chris-rock** [comment]
This is a great fix. Thanks for finding this edge case @jeremymv2
